racket/collects/pkg/main.rkt

```diff
@@ -154,651 +154,657 @@
-          #:install-copy-defns (install-copy-defns ...)
-          #:install-copy-checks (install-copy-checks ...))
-       (replace-context
-        stx
-         #`(commands
-            "This tool is used for managing installed packages."
-            "pkg-~a-command"
-            ;; ----------------------------------------
-            [install
-             "Install packages"
-             #:usage-help "Installs the packages specified by <pkg-source> ..., and"
-                          "if no sources are specified, installs the current directory"
-             #:once-each
-             install-type-flags ...
-             #:once-any
-             [install-dep-flags ...
-                                (dep-desc ...
-                                          install-dep-desc ...)]
-             [#:bool auto () "Shorthand for `--deps search-auto'"]
-             #:once-each
-             update-deps-flags ...
-             #:once-any
-             install-copy-flags/pre-clone ...
-             install-copy-flags/post-clone ...
-             #:once-any
-             scope-flags ...
-             #:multi
-             catalog-flags ...
-             #:once-each
-             [#:bool skip-installed () ("Skip a <pkg-source> if already installed")]
-             [#:bool pkgs () ("Install only the specified packages, even when none are provided")]
-             install-force-flags ...
-             install-clone-flags ...
-             dry-run-flags ...
-             job-flags ...
-             trash-flags ...
-             [#:bool fail-fast () ("Break `raco setup' when it discovers an error")]
-             #:args pkg-source
-             install-copy-defns ...
-             (let ([pkg-source
-                    ;; Implement special rules for an empty list of package sources
-                    (cond
-                     [(not (null? pkg-source))
-                      pkg-source]
-                     [clone
-                      (clone-to-package-name clone 'install)]
-                     [else
-                      pkg-source])])
-               (call-with-package-scope
-                'install
-                scope scope-dir installation user #f a-type #f name
-                (lambda ()
-                  install-copy-checks ...
-                  (when (and name (> (length pkg-source) 1))
-                    ((current-pkg-error) (format "the --name flag only makes sense with a single package source")))
-                  (unless (or (not name) (package-source->name name))
-                    ((current-pkg-error) (format "~e is an invalid package name" name)))
-                  ;; if no sources were supplied, and `--pkgs` was not
-                  ;; explicitly specified, install the current directory
-                  ;; as a linked directory
-                  (define-values (sources a-type*)
-                    (if (and (not pkgs) (null? pkg-source))
+(define-for-syntax (aliases names command-clause)
```

Comment:
This means the body will get duplicated, right? Is there a way to modify `commands` so that we don't need to duplicate the code?

Reply:
I think one way would be to have a procedure that is called instead, but the macro would still have to forward a lot of arguments (twice). I'll check, but I didn't see the body here being large.

Modifying commands to allow alias specs instead of make-commands might work; commands did a lot more work to expand into the (relatively undocumented?) svn-style command-line macro, and I didn't want to go too many layers deep.

Reply:
Also handled by the new version.